docs/source/nix-env.md

```diff
@@ -0,0 +1,46 @@
+# Nix-based Development Environment
+
+For the GDS backend flow, we use [Nix](https://nixos.org/) as our environment manager and development tool. Nix provides a reproducible, isolated environment for development and usage, ensuring that all dependencies are correctly managed. This is especially useful for complex EDA toolchains that have many dependencies and require specific versions of libraries and tools to function correctly. If you are only interested in using FABulous as a code generator and not in generating GDS files nor using [Librelane](https://github.com/librelane/librelane), you can use the Python virtual environment + OSS CAD suite installation method described in the [Usage](./Usage.rst).
+
+## Setting Up the Nix Environment
```

Comment:
Please also add a link to install nix manually. 
This could refer to the librelane docs. 

We should also add a point, what to do when someone has nix already installed.